test/e2e/suite/trustpolicy/trustpolicy_test.go

```diff
@@ -0,0 +1,467 @@
+package trustpolicy
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	. "github.com/notaryproject/notation/test/e2e/internal/notation"
+	"github.com/notaryproject/notation/test/e2e/internal/utils"
+	. "github.com/notaryproject/notation/test/e2e/suite/common"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestTrustPolicy(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Trust Policy Suite")
+}
+
+var _ = Describe("notation trust policy", func() {
+	It("test empty registryScope", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			// update trustpolicy.json
+			vhost.SetOption(AddTrustPolicyOption("empty_registry_scope_trustpolicy.json"))
+
+			// test localhost:5000/test-repo
+			OldNotation().Exec("sign", artifact.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest()).
+				MatchErrKeyWords("trust policy statement \"e2e\" has zero registry scopes")
+		})
+	})
+
+	It("test registryScope with a repository", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			// update trustpolicy.json
+			vhost.SetOption(AddTrustPolicyOption("registry_scope_trustpolicy.json"))
+
+			// generate an artifact with given repository name
+			artifact := GenerateArtifact("", "test-repo")
+
+			// test localhost:5000/test-repo
+			OldNotation().Exec("sign", artifact.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.Exec("verify", artifact.ReferenceWithDigest()).MatchKeyWords(VerifySuccessfully)
+		})
+	})
+
+	It("test registryScope with multiple repositories", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			// update trustpolicy.json
+			vhost.SetOption(AddTrustPolicyOption("multiple_registry_scope_trustpolicy.json"))
+
+			// generate an artifact with given repository name
+			artifact2 := GenerateArtifact("", "test-repo2")
+			artifact3 := GenerateArtifact("", "test-repo3")
+
+			// test localhost:5000/test-repo2
+			OldNotation().Exec("sign", artifact2.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.Exec("verify", artifact2.ReferenceWithDigest()).MatchKeyWords(VerifySuccessfully)
+
+			// test localhost:5000/test-repo3
+			OldNotation().Exec("sign", artifact3.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.Exec("verify", artifact3.ReferenceWithDigest()).MatchKeyWords(VerifySuccessfully)
+		})
+	})
+
+	It("test registryScope with any(*) repository", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
```

Comment:
Which trust policy file is this test using? Can we explicitly configure this to "any_registry_scope_trust_policy.json"?

Reply:
Updated to specify `any_registry_scope_trust_policy.json`.